### ISSUE TITLE
[JS-Controller 6] rflink adapter does not start

### ISSUE THREAD

**Issue by mcm1957** [ioBroker/ioBroker.rflink#48]
Adapter does not start with js-controller 6. Most likely adapter-core must be updated.

```
2024-06-10 22:13:58.617	error	Caught by controller[0]: /opt/iobroker/node_modules/iobroker.rflink/main.js:43
host.raspberrypi
	2024-06-10 22:13:58.618	error	Caught by controller[0]: adapter = new utils.Adapter(options);
host.raspberrypi
	2024-06-10 22:13:58.618	error	Caught by controller[0]: ^
host.raspberrypi
	2024-06-10 22:13:58.619	error	Caught by controller[0]: TypeError: utils.Adapter is not a constructor
host.raspberrypi
	2024-06-10 22:13:58.623	error	Caught by controller[0]: at startAdapter (/opt/iobroker/node_modules/iobroker.rflink/main.js:43:15)
host.raspberrypi
	2024-06-10 22:13:58.623	error	Caught by controller[0]: at Object.<anonymous> (/opt/iobroker/node_modules/iobroker.rflink/main.js:717:5)
host.raspberrypi
	2024-06-10 22:13:58.624	error	Caught by controller[0]: at Module._compile (node:internal/modules/cjs/loader:1364:14)
host.raspberrypi
	2024-06-10 22:13:58.624	error	Caught by controller[0]: at Module._extensions..js (node:internal/modules/cjs/loader:1422:10)
host.raspberrypi
	2024-06-10 22:13:58.624	error	Caught by controller[0]: at Module.load (node:internal/modules/cjs/loader:1203:32)
host.raspberrypi
	2024-06-10 22:13:58.625	error	Caught by controller[0]: at Module._load (node:internal/modules/cjs/loader:1019:12)
host.raspberrypi
	2024-06-10 22:13:58.625	error	Caught by controller[0]: at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:128:12)
host.raspberrypi
	2024-06-10 22:13:58.625	error	Caught by controller[0]: at node:internal/main/run_main_module:28:49
host.raspberrypi
	2024-06-10 22:13:58.625	error	Caught by controller[0]: Node.js v18.20.3
host.raspberrypi
	2024-06-10 22:13:58.626	error	instance system.adapter.rflink.0 terminated with code 1 (JS_CONTROLLER_STOPPED)
```

**Comment by mcm1957** [comment]
should be fixed with 2.1.7 - BUT does not seem so ...

https://forum.iobroker.net/topic/75222/rflink-keine-serielle-schnittstelle/14

```
host.GLT
2024-06-11 20:10:24.709	info	Rebuild for adapter system.adapter.rflink.0 not successful in 3 tries. Adapter will not be restarted again. Please execute "npm install --production" in adapter directory manually.
host.GLT
2024-06-11 20:10:24.709	error	instance system.adapter.rflink.0 terminated with code 1 (JS_CONTROLLER_STOPPED)
host.GLT
2024-06-11 20:10:24.709	error	Caught by controller[1]: Node.js v20.14.0
host.GLT
2024-06-11 20:10:24.709	error	Caught by controller[1]:
host.GLT
2024-06-11 20:10:24.709	error	Caught by controller[1]: at node:internal/main/run_main_module:28:49
host.GLT
2024-06-11 20:10:24.709	error	Caught by controller[1]: at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:174:12)
host.GLT
2024-06-11 20:10:24.709	error	Caught by controller[1]: at Module._load (node:internal/modules/cjs/loader:1024:12)
host.GLT
2024-06-11 20:10:24.709	error	Caught by controller[1]: at Module.load (node:internal/modules/cjs/loader:1208:32)
host.GLT
2024-06-11 20:10:24.709	error	Caught by controller[1]: at Module._extensions..js (node:internal/modules/cjs/loader:1416:10)
host.GLT
2024-06-11 20:10:24.709	error	Caught by controller[1]: at Module._compile (node:internal/modules/cjs/loader:1358:14)
host.GLT
2024-06-11 20:10:24.709	error	Caught by controller[1]: at Object.<anonymous> (C:\ioBroker\GLT\node_modules\iobroker.rflink\main.js:17:17)
host.GLT
2024-06-11 20:10:24.709	error	Caught by controller[1]: TypeError: Cannot read properties of undefined (reading 'supportsFeature')
host.GLT
2024-06-11 20:10:24.709	error	Caught by controller[1]:
host.GLT
2024-06-11 20:10:24.709	error	Caught by controller[1]: ^
host.GLT
2024-06-11 20:10:24.709	error	Caught by controller[1]: if (adapter.supportsFeature && !adapter.supportsFeature('CONTROLLER_NPM_AUTO_REBUILD')) {
host.GLT
2024-06-11 20:10:24.709	error	Caught by controller[1]: C:\ioBroker\GLT\node_modules\iobroker.rflink\main.js:17
host.GLT
2024-06-11 20:10:24.709	error	Caught by controller[0]: → C:\ioBroker\GLT\node_modules\@serialport\bindings\lib\binding\node-v115-win32-x64\bindings.node
host.GLT
2024-06-11 20:10:24.709	error	Caught by controller[0]: → C:\ioBroker\GLT\node_modules\@serialport\bindings\addon-build\default\install-root\bindings.node
host.GLT
2024-06-11 20:10:24.709	error	Caught by controller[0]: → C:\ioBroker\GLT\node_modules\@serialport\bindings\addon-build\debug\install-root\bindings.node
host.GLT
2024-06-11 20:10:24.708	error	Caught by controller[0]: → C:\ioBroker\GLT\node_modules\@serialport\bindings\addon-build\release\install-root\bindings.node
host.GLT
2024-06-11 20:10:24.708	error	Caught by controller[0]: → C:\ioBroker\GLT\node_modules\@serialport\bindings\compiled\20.14.0\win32\x64\bindings.node
host.GLT
2024-06-11 20:10:24.708	error	Caught by controller[0]: → C:\ioBroker\GLT\node_modules\@serialport\bindings\build\default\bindings.node
host.GLT
2024-06-11 20:10:24.708	error	Caught by controller[0]: → C:\ioBroker\GLT\node_modules\@serialport\bindings\Release\bindings.node
host.GLT
2024-06-11 20:10:24.708	error	Caught by controller[0]: → C:\ioBroker\GLT\node_modules\@serialport\bindings\out\Release\bindings.node
host.GLT
2024-06-11 20:10:24.708	error	Caught by controller[0]: → C:\ioBroker\GLT\node_modules\@serialport\bindings\Debug\bindings.node
host.GLT
2024-06-11 20:10:24.708	error	Caught by controller[0]: → C:\ioBroker\GLT\node_modules\@serialport\bindings\out\Debug\bindings.node
host.GLT
2024-06-11 20:10:24.708	error	Caught by controller[0]: → C:\ioBroker\GLT\node_modules\@serialport\bindings\build\Release\bindings.node
host.GLT
2024-06-11 20:10:24.708	error	Caught by controller[0]: → C:\ioBroker\GLT\node_modules\@serialport\bindings\build\Debug\bindings.node
host.GLT
2024-06-11 20:10:24.708	error	Caught by controller[0]: → C:\ioBroker\GLT\node_modules\@serialport\bindings\build\bindings.node
host.GLT
2024-06-11 20:10:24.707	error	Caught by controller[0]: Cannot load serialport module : Error: Could not locate the bindings file. Tried:
```

**Comment by Apollon77** [comment]
the serialport nerror requires serialpoert upgrade ... 8. x is way oiutdated for current nodejs versions. But the supportsFeature error is strange

**Comment by foxriver76** [comment]
fixed with 2.1.8
